### PR TITLE
[Echo] Push notification e2e ui test

### PR DESCRIPTION
**Pull request by radeknovis**
# Description
Adds UI test which 
- creates new pairing
-  fires off push notification
- asserts PN is showed

It sits in an entirely new UI test target which would allow running the test as part of the Echo server pipeline

Resolves # (issue)
https://github.com/WalletConnect/WalletConnectSwiftV2/issues/782

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Example/WalletApp/ApplicationLayer/AppDelegate.swift [hunk 1]
```diff
@@ -24,14 +33,16 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
...
-        Task(priority: .high) {
-            // Use pasteboard for testing purposes
-            let tokenParts = deviceToken.map { data in String(format: "%02.2hhx", data) }
-            let token = tokenParts.joined()
-            let pasteboard = UIPasteboard.general
-            pasteboard.string = token
-            try await Push.wallet.register(deviceToken: deviceToken)
-        }
+        
+       // Commenting this out as it breaks UI tests that copy/paste URI
```
Comment (radeknovis) [inline Example/WalletApp/ApplicationLayer/AppDelegate.swift:37]: Question for reviewers...
Not sure how much this is used and whether I should keep it and devise a workaround.

File: Example/WalletApp/PresentationLayer/Wallet/Welcome/WelcomePresenter.swift [hunk 2]
```diff
@@ -16,9 +16,11 @@ final class WelcomePresenter: ObservableObject {
...
     
     func onGetStarted() {
-        let pasteboard = UIPasteboard.general
-        let clientId = try? Networking.interactor.getClientId()
-        pasteboard.string = clientId
+        
+        // Commenting this out as it breaks UI tests that copy/paste URI
+//        let pasteboard = UIPasteboard.general
+//        let clientId = try? Networking.interactor.getClientId()
+//        pasteboard.string = clientId
```
Comment (radeknovis) [inline Example/WalletApp/PresentationLayer/Wallet/Welcome/WelcomePresenter.swift:23]: Question for reviewers...
Not sure how much this is used and whether I should keep it and devise a workaround.

File: Example/DApp/AppDelegate.swift [hunk 3]
```diff
@@ -4,6 +4,16 @@ import UIKit
 class AppDelegate: UIResponder, UIApplicationDelegate {
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         // Override point for customization after application launch.
+        
+        #if DEBUG
+        if ProcessInfo.processInfo.arguments.contains("-disableAnimations") {
```
Comment (radeknovis) [inline Example/DApp/AppDelegate.swift:9]: Maybe this is overkill for single test

File: Example/WalletApp/ApplicationLayer/AppDelegate.swift [hunk 4]
```diff
@@ -24,12 +24,15 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
       _ application: UIApplication,
       didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
     ) {
+
         Task(priority: .high) {
+            // Commenting this out as it breaks UI tests that copy/paste URI
```
Comment (flypaper0) [inline Example/WalletApp/ApplicationLayer/AppDelegate.swift:29]: May be just remove? What the reason of doing it?

**Review by flypaper0 — APPROVED** [review]

File: Example/WalletApp/PresentationLayer/Wallet/Welcome/WelcomePresenter.swift (hunk 2 above)
Comment (llbartekll) [reply Example/WalletApp/PresentationLayer/Wallet/Welcome/WelcomePresenter.swift:23]: or any other way you like but we need to be able to get it. It's sometimes important to check the relay logs for particular client id.

**Review by llbartekll — COMMENTED** [review]
Looks good to me, just a few comments

**Pull request merged**